### PR TITLE
Override blacklight's thumbnail methods to provide a default for bibliography items

### PR DESCRIPTION
**Pull request by jkeck**
Connects to #651 

## List View
list-view

## Admin View
admin-view

## Masonry View
masonry-view

## Gallery View
gallery-view

I didn't add the Slideshow view because @blalbrit said we could disable that view.

### CODE REVIEW

**Comment by mejackreed** [comment]
Not a review, but just a comment. I like the approach of #2 better as it feels "more right". But the simplicity of approach #1 is hard to argue with.

Have you looked into which methods need to be overridden for approach #2? Does that live in Spotlight or somewhere else?

**Comment by jkeck** [comment]
It's not so much overridden, it is just that we'll need to configure `thumbnail_method` in the view configuration.  That method will effectively be something along the lines of:

```ruby
def index_thumbnail_method(*args)
  if document.reference?
    image_tag('fallback-image')
  else
    blacklight_render_method(some_subset_of_args)
  end
end
```

Also, because of the different possible thumbnails used in different view types we would either need to provide different methods for different views or provide logic w/i the method to point to the appropriate sized thumbnail.

File: lib/traject/bibtex_config.rb [hunk 1]
```diff
@@ -116,6 +116,18 @@
...
+}
+
+to_field 'thumbnail_url_ssm', lambda { |_, accumulator, _|
+  accumulator << ActionController::Base.helpers.asset_path('default-thumbnail-book.png')
+}
+
+to_field 'large_image_url_ssm', lambda { |_, accumulator, _|
+  accumulator << ActionController::Base.helpers.asset_path('default-thumbnail-book.png')
+}
+
```
Comment (drh-stanford) [inline lib/traject/bibtex_config.rb]: I think you can do these with the `literal` macro, namely:

```
to_field 'large_image_url_ssm', literal(ActionController::Base.helpers.asset_path('default-thumbnail-book.png').to_s)
```

**Comment by mejackreed** [comment]
Gotcha thanks @jkeck for that clarification.

Perhaps with your comment in mind, I'm more inclined to lean towards option #2. For the following reasons:
 - It doesn't force us to index an asset path for display purposes only (this seems suboptimal)
 - It allows us to tap into (perhaps in the correct seam?) the ability for falling back when there is no indexed thumb. This seems like it could be desirable.
 - Would potentially force a reindex in production for an unknown exhibit just to get access to a new icon/image.

What do you think?

**Pull request merged**